### PR TITLE
fix: Fix voice commands actions in voice prompt window

### LINKED ISSUE #2171: Go to chat Activity as soon as voice command is pressed
**Describe the bug**
Currently, the voice commands even after clicked waits until the speech recognition timesout.
This shouldn't happen. It should go to the ChatActivity otherwise it doesn't justify the purpose of these commands. 

**Expected Behaviour**
The voice commands should go to chatAcitivity and display results as soon as they are clicked.

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by atm1504**
Fixes #2171 #2241 

Screenshots for the change: 
ezgif com-video-to-gif

### CODE REVIEW

**Comment by atm1504** [comment]
@arundhati24 @batbrain7 @iamareebjamal please review it.

**Comment by iamareebjamal** [comment]
First of all, tell me who has told you that restarting an activity with FLAG_ACTIVITY_CLEAR_TOP is a good idea and solution of the bug. From next time on, if your solution involves FLAG_ACTIVITY_CLEAR_TOP, just don't send the PR and think of other solution

**Comment by iamareebjamal** [comment]
This is like "Oh, I have to show a new message SUSI sent from the server and it is not displaying. Let me just restart the activity so that all messages are refetched from the server and the new message will be displayed. Easy Peasy"

What a terrible solution

Revert all PRs you have made involving FLAG_ACTIVITY_CLEAR_TOP

**Comment by atm1504** [comment]
@iamareebjamal @arundhati24 @batbrain7 please review it. Commit 51f8b67 is a different method that has been used and it's working fine.

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt [hunk 1]
```diff
@@ -32,13 +37,16 @@ class VoiceCommandsAdapter(val items: ArrayList<String>, val context: Context?)
...
+            voiceCommand.setOnClickListener {
+                val message = items[adapterPosition]
+                val this_activity = context
+                (this_activity as ChatActivity).setText(message)
+                this_activity.searchChat.show()
+                this_activity.voiceSearchChat.show()
+                this_activity.fabsetting.show()
+                this_activity.btnSpeak.isEnabled = true
+                chatPresenter.startHotwordDetection()
+                this_activity.supportFragmentManager.popBackStackImmediate()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt]: Why not sending to presenter

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt]: I tried that also. But, what I found that, as we are not starting the chatActivity (none of the following actions onResume, onPause, onCreate gets executed), so the presenter is failing to control any UI changes of the chatActivty. Hence, that method failed. Then, I found out this method useful.

Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt]: This doesn't seem to be a good way to set the click listener. To keep things clean, you should create and pass an item click listener (that requires adapter position and the message) from the view to the adapter, while initialising the adapter, and inside the init of view holder provide the adapter position and message to be set. Then, from your fragment, tell your presenter to start hotword detection and once that is done successfully, let the presenter tell the view to display any changes, that are required, to the user.

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt]: @arundhati24 I even tried that method. Whats, the issue here is that when we do popBackStackImmediate(), the ChatActivity is opened. The issue here is, the state of ChatActivity is neither onPause, onResume or onCreate. So, even if I try to do anything using view, it simply fails. Those options that have onClick function in XML only work. Even, the chat presenter doesn't get access to the view in such cases(I have checked it). That was the main reason, I came up with this kind of solution, after so many tries. I also found, that the one who created the files of STT Fragment, also had a similar mindset.

**Pull request merged**